awx/main/tests/functional/conftest.py

```diff
@@ -351,9 +352,27 @@ def kube_credential(credentialtype_kube):
     )
 
 
+# @pytest.fixture
+# def inventory(organization):
+#     return organization.inventories.create(name="test-inv")
+
+
 @pytest.fixture
 def inventory(organization):
-    return organization.inventories.create(name="test-inv")
+    available_providers = tuple(compute_cloud_inventory_sources())
+
+    if available_providers:
+        if not organization.inventories.exists():
+            organization.inventories.create(name="test-inv")
+
+        first_inventory = organization.inventories.first()
+
+        if first_inventory:
+            return first_inventory
+        else:
+            raise ValueError("No inventories found in the queryset")
+    else:
+        raise ValueError("No available cloud providers found")
```

Comment:
Let's reduce the indentation where possible:
```suggestion

        raise ValueError("No inventories found in the queryset")

    raise ValueError("No available cloud providers found")
```